include/solvers/petsc_nonlinear_solver.h

```diff
@@ -153,6 +158,18 @@ class PetscNonlinearSolver : public NonlinearSolver<T>
    */
   void use_default_monitor(bool state) { _default_monitor = state; }
 
+  /**
+   * Petsc solve type
+   */
+  enum SolveType
+  {
+    MF_OPERATOR, ///< Preconditioned Jacobian-free Newton
+    MF,          ///< Jacobian-free Newton
+    STANDARD     ///< Newton with explicit matrix
+  };
+
+  SolveType solve_type;
```

Comment:
I think I'd prefer a protected variable behind a setter/getter interface, to be consistent with the rest of the `PetscNonlinearSolver` interface...

Reply:
Seconded.